### ISSUE TITLE
ENH: convert classify-samples-ncv to a pipeline, or add NCV pipeline 

### ISSUE THREAD

**Issue by nbokulich** [qiime2/q2-sample-classifier#160]
**Improvement Description**
convert classify-samples-ncv to a pipeline, or add NCV pipeline. This will reduce user effort while utilizing these methods.

**Current Behavior**
The *-ncv actions are methods, not pipelines. Downstream visualizers are run separately by users.

**Proposed Behavior**
Stitch together *-ncv pipelines, either by converting the existing methods or by adding new pipelines.

**Comment by Oddant1** [comment]
So this would be two new pipelines one with `regress_samples_ncv` and one with `classify_samples_ncv`, what else would be in these pipelines? Just a visualizer? What visualizer? And we could always be kinda lazy and just call them `regress_samples_ncv_pipeline` and so on.

**Comment by nbokulich** [comment]
yep, two pipelines one with regress_samples_ncv and one with classify_samples_ncv.

the rest of the pipelines would look a lot like what `regress_samples` and `classify_samples` do after training. Something like this:
1. `*_samples_ncv`
2. send predictions to the appropriate visualizer: `scatterplot` or `confusion_matrix`
3. `classify_samples_ncv` could make a heatmap of the top features.

**Comment by Oddant1** [comment]
@nbokulich The more I read your comment, the more I feel like I'm missing something. Let's say we're working with `classify_samples_ncv`. I read your above comment as

1. call the current `classify_samples_ncv` method from the new `classify_samples_ncv_pipeline` or whatever we call it pipeline
2. use the results of the call to `classify_samples_ncv` to call the appropriate viz from the pipeline
3. use the results of the prior two calls to generate a heatmap from the pipeline
4. return all relevant results from the pipeline

**Comment by nbokulich** [comment]
yes. I think all you are missing is that there are some pipeline differences between `classify_samples` and `regress_samples` that we would want to replicate in a potential NCV pipeline. So, e.g., the heatmap would only be generated for `classify_samples_ncv_pipeline`, not `regress...`. But otherwise it sounds like we are on the same page.
